services/src/test/scala/cromwell/services/metadata/MetadataServiceSpec.scala

```diff
@@ -23,41 +23,109 @@ class MetadataServiceSpec extends FlatSpec with Matchers with TableDrivenPropert
+    womValueToMetadataEvents(MetadataKey(workflowId, None, "root"), emptyWomMap).toList should contain theSameElementsAs List(
       MetadataEvent.empty(MetadataKey(workflowId, None, "root"))
     )
   }
 
+  it should "convert a WomMaybePopulatedFile to MetadataEvents" in {
+    import MetadataService._
+    val workflowId = WorkflowId.randomId()
+    // Leave some fields out on purpose to verify that they still make it (empty) to the metadata
+    val womMaybePopulatedFileInner = WomMaybePopulatedFile(
+      valueOption = Option("innerFileValue"),
+      contentsOption = Option("innerContents")
+    )
+
+    val womMaybePopulatedFileOuter = WomMaybePopulatedFile(
+      valueOption = Option("fileValue"),
+      checksumOption = Option("checksum"),
+      sizeOption = Option(5L),
+      formatOption = Option("format"),
+      contentsOption = Option("contents"),
+      secondaryFiles = List(
+        womMaybePopulatedFileInner,
+        WomSingleFile("outerSingleFile")
+      )
+    )
+
+    womValueToMetadataEvents(MetadataKey(workflowId, None, "root"), womMaybePopulatedFileOuter).toList should contain theSameElementsInOrderAs List(
+      MetadataEvent(MetadataKey(workflowId, None, "root:class"), MetadataValue("File")),
+      MetadataEvent(MetadataKey(workflowId, None, "root:location"), MetadataValue("fileValue")),
+      MetadataEvent(MetadataKey(workflowId, None, "root:checksum"), MetadataValue("checksum")),
+      MetadataEvent(MetadataKey(workflowId, None, "root:size"), MetadataValue(5L)),
+      MetadataEvent(MetadataKey(workflowId, None, "root:format"), MetadataValue("format")),
+      MetadataEvent(MetadataKey(workflowId, None, "root:contents"), MetadataValue("contents")),
+      MetadataEvent(MetadataKey(workflowId, None, "root:secondaryFiles[0]:class"), MetadataValue("File")),
+      MetadataEvent(MetadataKey(workflowId, None, "root:secondaryFiles[0]:location"), MetadataValue("innerFileValue")),
+      MetadataEvent(MetadataKey(workflowId, None, "root:secondaryFiles[0]:checksum"), MetadataValue("", MetadataNull)),
+      MetadataEvent(MetadataKey(workflowId, None, "root:secondaryFiles[0]:size"), MetadataValue("", MetadataNull)),
+      MetadataEvent(MetadataKey(workflowId, None, "root:secondaryFiles[0]:format"), MetadataValue("", MetadataNull)),
+      MetadataEvent(MetadataKey(workflowId, None, "root:secondaryFiles[0]:contents"), MetadataValue("innerContents")),
+      MetadataEvent(MetadataKey(workflowId, None, "root:secondaryFiles[0]:secondaryFiles[]"), None),
+      MetadataEvent(MetadataKey(workflowId, None, "root:secondaryFiles[1]"), MetadataValue("outerSingleFile"))
+    )
+  }
+
+  it should "convert a WomMaybeListedDirectory to MetadataEvents" in {
+    import MetadataService._
+    val workflowId = WorkflowId.randomId()
+    // Leave some fields out on purpose to verify that they still make it (empty) to the metadata
```

Comment:
What do they look like in metadata? Just a `null`?

Reply:
yes

Reply:
except `secondaryFiles` and `listing` I should say which being arrays will be empty arrays instead